### DISCUSSION TITLE
PICO W and RA8875 configuration issues

### DISCUSSION THREAD

**Discussion by 32-bits** [lovyan03/LovyanGFX#618] [Device Configurations]
Hi, I would like to configure LGFX for a Pico W and RA8875 7" display using SPI. The display is connected to the default SPI pins on the Pico and works correctly using the Adafruit RA8875 driver and GFX library, I can display graphics and basic text, so the connections are correct. For LGFX I tried Arduino and Platformio, both generate lots of errors, which I assume are configuration errors, but my knowledge of coding is not good enough to fix these. Most of the errors make references to functions or items not declared, so maybe there is a declaration file missing?  Is there a simple walk through on how to add the RA8875 display and configure for the PICO W? 

A basic walk through might also be helpful to other, less knowledgeable users, who would like to use the LGFX libruary.

Thanks Michael

**Comment by tobozo** [comment]
Hi, thanks for your feedback :+1: 

you may want to try the [develop](https://github.com/lovyan03/LovyanGFX/tree/develop) branch of this repository, it contains fixes for the latest pico sdk.

**Comment by 32-bits** [comment]
I have tried to compile the example at https://lovyangfx.readthedocs.io/en/latest/02_using.html using platformio and I get the following errors. Can you help explain what I am doing wrong?

 *  Executing task: C:\Users\mh_dv\.platformio\penv\Scripts\platformio.exe run 

Processing pico (platform: https://github.com/maxgerhardt/platform-raspberrypi.git; board: pico; framework: arduino)
--------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/raspberrypi/pico.html
PLATFORM: Raspberry Pi RP2040 (1.14.0+sha.9e55f6d) > Raspberry Pi Pico
HARDWARE: RP2040 133MHz, 256KB RAM, 2MB Flash
DEBUG: Current (blackmagic) External (blackmagic, cmsis-dap, jlink, pico-debug, picoprobe, raspberrypi-swd)
PACKAGES: 
 - framework-arduinopico @ 1.40002.0+sha.aaaa99d 
 - tool-picotool-rp2040-earlephilhower @ 5.120300.240827 (12.3.0)
 - toolchain-rp2040-earlephilhower @ 5.120300.240125 (12.3.0)
Flash size: 2.00MB
Sketch size: 2.00MB
Filesystem size: 0.00MB
Maximium Sketch size: 2093056 EEPROM start: 0x101ff000 Filesystem start: 0x101ff000 Filesystem end: 0x101ff000
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 66 compatible libraries
Scanning dependencies...
Dependency Graph
|-- LovyanGFX @ 1.1.16
Building in release mode
Compiling .pio\build\pico\libf39\LovyanGFX\lgfx\v1\platforms\rp2040\common.cpp.o
Compiling .pio\build\pico\src\main.cpp.o
.pio\libdeps\pico\LovyanGFX\src\lgfx\v1\platforms\rp2040\common.cpp:78:56: error: use of enum 'gpio_function' without previous declaration
   78 |     bool lgfx_gpio_set_function(int_fast16_t pin, enum gpio_function fn)
      |                                                        ^~~~~~~~~~~~~
.pio\libdeps\pico\LovyanGFX\src\lgfx\v1\platforms\rp2040\common.cpp: In function 'bool lgfx::v1::{anonymous}::lgfx_gpio_set_function(int_fast16_t, int)':
.pio\libdeps\pico\LovyanGFX\src\lgfx\v1\platforms\rp2040\common.cpp:92:16: error: 'iobank0_hw_t' does not name a type; did you mean 'io_bank0_hw_t'?
   92 |       volatile iobank0_hw_t *iobank0_regs = reinterpret_cast<volatile iobank0_hw_t *>(IO_BANK0_BASE); 
      |                ^~~~~~~~~~~~
      |                io_bank0_hw_t
.pio\libdeps\pico\LovyanGFX\src\lgfx\v1\platforms\rp2040\common.cpp:93:7: error: 'iobank0_regs' was not declared in this scope; did you mean 'iobank0_hw'?
   93 |       iobank0_regs->io[pin].ctrl = fn << IO_BANK0_GPIO0_CTRL_FUNCSEL_LSB;
      |       ^~~~~~~~~~~~
      |       iobank0_hw
*** [.pio\build\pico\libf39\LovyanGFX\lgfx\v1\platforms\rp2040\common.cpp.o] Error 1
src\main.cpp:39:8: error: 'LGFX' does not name a type
   39 | static LGFX lcd;                 // Instance of LGFX
      |        ^~~~
src\main.cpp:40:28: error: 'lcd' was not declared in this scope
   40 | static LGFX_Sprite sprite(&lcd); // Instance of LGFX_Sprite when using sprites
      |                            ^~~
src\main.cpp: In function 'void setup()':
src\main.cpp:58:3: error: 'lcd' was not declared in this scope
   58 |   lcd.init();
      |   ^~~
src\main.cpp: In function 'void loop()':
src\main.cpp:351:19: error: 'lcd' was not declared in this scope
  351 |   if ((x += 2) >= lcd.width()) x -= lcd.width();
      |                   ^~~
src\main.cpp:352:19: error: 'lcd' was not declared in this scope
  352 |   if ((y += 1) >= lcd.height()) y -= lcd.height();
      |                   ^~~
src\main.cpp:353:29: error: 'lcd' was not declared in this scope
  353 |   sprite.setPaletteColor(1, lcd.color888( 0, 0, count & 0xFF));
      |                             ^~~
*** [.pio\build\pico\src\main.cpp.o] Error 1
========================================= [FAILED] Took 7.70 seconds =========================================
